File: Sample/OBJ-C/Sample/Sample/ViewModels/SecondViewModel.m
//
//  SecondViewModel.m
//  Sample
//
//  Created by Dabonot Francois on 20/01/2018.
//  Copyright © 2018 Dabonot Francois. All rights reserved.
//

#import "SecondViewModel.h"

@implementation SecondViewModel

- (void)closeViewModel
{
    NSLog(@"closeViewModel start!");
    [[Locator get:@protocol(INavigationService)] closeCurrentViewModelOnCompletion:^{
        NSLog(@"closeViewModel end!");
    }];
}

@end

File: Sample/Swift/Sample/Sample/ViewModels/SecondViewModel.swift
//
//  SecondViewModel.swift
//  Sample
//
//  Created by Dabonot Francois on 20/01/2018.
//  Copyright © 2018 Dabonot Francois. All rights reserved.
//

import UIKit

class SecondViewModel: BaseViewModel  {
    
    func closeViewModel() -> Void {
        NSLog("closeViewModel start!")
        (Locator.get(INavigationService.self) as! INavigationService).closeCurrentViewModel(onCompletion: {
            NSLog("closeViewModel over!")
        });
    }

}
